### PR TITLE
Initial Table Drop

### PR DESCRIPTION
**Pull request by tay1orjones**
**Summary**

- initial lightweight table, pagination, toolbar from @bryancboyd
- my first pass at table filtering

**Change List (commits, features, bugs, etc)**

- initial table port from internal platform component repo
- add filter header row
- replace deprecated immutability addon with `immutability-helper`
- increase code coverage across all table components

**Related Issues**

- #10 
- #7 

**Acceptance Test (how to verify the PR)**

- updates are visible in storybook

### CODE REVIEW

**Comment by tay1orjones** [comment]
There is definitely more work here to do, but I figured we could get this baseline drop in as it stands.

**Comment by bryancboyd** [comment]
I've added some improvements to the branch, but did a commit without verification... there were a few snapshot errors I wasn't sure how to resolve.  Here's the how it looks:

image

**Comment by tay1orjones** [comment]
@scottdickerson this is ready for review.

- [x] Fix pagination functionality
- [x] Update/fix tests for filtering and pagination
- [x] Add filter toggle button in toolbar

File: package.json [hunk 1]
```diff
@@ -30,6 +30,9 @@
     }
   },
   "dependencies": {
+    "carbon-components": "9.66.3",
```
Comment (scottdickerson) [inline package.json]: we want these only in dev-dependencies

File: src/utils/componentUtilityFunctions.js [hunk 2]
```diff
@@ -12,3 +12,14 @@ export function scrollErrorIntoView(focus = true) {
...
 }
+
+export const handleEnterKeyDown = (evt, callback) => {
+  if (evt.key === 'Enter') {
+    callback(evt);
+  }
+};
+
+export const defaultFunction = name => {
+  console.log(this);
```
Comment (scottdickerson) [inline src/utils/componentUtilityFunctions.js]: this line is causing us some compilation warnings, so going to remove for now.

File: package.json (hunk 1 above)
Comment (tay1orjones) [reply package.json]: oop, thanks for this. was likely a leftover from the yarn issues we were having during our pairing session

**Pull request merged**